src/backend/distributed/executor/multi_router_executor.c

```diff
@@ -751,3 +902,135 @@ RouterExecutorEnd(QueryDesc *queryDesc)
 	queryDesc->estate = NULL;
 	queryDesc->totaltime = NULL;
 }
+
+
+/* TODO: write comment */
+void
+InstallRouterExecutorShmemHook()
+{
+	prev_shmem_startup_hook = shmem_startup_hook;
+	shmem_startup_hook = RegisterRouterExecutorXactCallbacks;
+}
+
+
+/* TODO: Write comment */
+static void
+RegisterRouterExecutorXactCallbacks()
+{
+	RegisterXactCallback(RouterTransactionCallback, NULL);
```

Comment:
Need to make sure whether work already has performed here. On windows / EXEC_BACKEND environments the startup hooks get called on each backend startup (since there's no fork to copy state).